src/PermitTokens/permitSignature.ts

```diff
@@ -0,0 +1,131 @@
+import { Wallet } from "ethers";
+
+async function getCorrectPermitSig(
+    wallet: Wallet,
+    token: any,
+    spender: string,
+    value: any,
+    deadline: any,
+    optional?: { nonce?: number; name?: string; chainId?: number; version?: string }
+    ) { 
+    const [nonce, name, version, chainId] = await Promise.all([
+        optional?.nonce ?? token.nonces(wallet.address),
+        optional?.name ?? token.name(),
+        optional?.version ?? '1',
+        optional?.chainId ?? wallet.getChainId(),
+    ])
+    
+    const domain = {
+        "name": name,
+        "version": version,
+        "chainId": chainId,
+        "verifyingContract": token.address
+    };
```

Comment:
easier to query `DOMAIN_SEPARATOR()` onchain
https://eips.ethereum.org/EIPS/eip-2612#specification

Reply:
when I do this certain contracts throw an error like uni, don't think all of them have domain_separator()

Reply:
hm seems like uni calculates it on the fly
```solidity
bytes32 domainSeparator = keccak256(abi.encode(DOMAIN_TYPEHASH, keccak256(bytes(name)), getChainId(), address(this)));
```